siobrultech_protocols/gem/protocol.py

```diff
@@ -120,5 +141,103 @@ def skip_malformed_packet(msg, *args, **kwargs):
             except MalformedPacketException as e:
                 skip_malformed_packet(e.args[0])
 
+        self._ensure_transport()
+
+    def _ensure_transport(self) -> asyncio.BaseTransport:
         if not self._transport:
             raise EOFError
+
+        return self._transport
+
+
+class ProtocolState(Enum):
```

Comment:
We should do [`@unique`](https://docs.python.org/3/library/enum.html#enum.unique) on this `enum`.